cli/units_cmd.go

```diff
@@ -111,6 +118,22 @@ func scanUnitsIntoConfig(cfg *config.Repository, quiet bool) error {
 	return nil
 }
 
+// checkRegularExecutableFile verifies that the file pointed to by name is a regular executable file.
+func checkRegularExecutableFile(name string) error {
+	reason := error(nil)
+	switch info, err := os.Stat(name); {
+	case err != nil:
+		reason = fmt.Errorf("caused %s", err)
+	case info.IsDir():
+		reason = fmt.Errorf("a directory")
+	case !info.Mode().IsRegular():
+		reason = fmt.Errorf("not a regular file")
+	case info.Mode().Perm()&0111 == 0:
+		reason = fmt.Errorf("not an executable file")
+	}
+	return fmt.Errorf("%s should be a regular executable file, but it's %s", name, reason)
```

Comment:
This is okay, but if it were me, I wouldn't be afraid to copy the `return fmt.Errorf("%s should be a regular executable file`  part 5 times.

When it comes to DRY, it's valuable to avoid duplication of high level ideas and concepts. But duplicating a few lines of code in a small context where it's clear what's going on is very inexpensive.

Also see Rob Pike's "A little copying is better than a little dependency." from https://go-proverbs.github.io/.